### PR TITLE
  Add context in functions to make debugging easier and Add GetBuildFromQueueID() function

### PR DESCRIPTION
**Pull request by figo**
1) Add context in functions to make debugging easier

    if "debug" been set in context, will dump response data including body.
    we can pass logger in context in the future

2) Add GetBuildFromQueueID() function

    a) This function going to return the build object base on task/queue id.
    Jenkins will queue a task for a few second and assign the build number.

    b) Update REAME file for use case of triggering a job and get the result of the build.
    Update README with context paramete

    Signed-off-by: Hui Luo <luoh@vmware.com>

### CODE REVIEW

File: artifact.go [hunk 1]
```diff
@@ -50,19 +51,19 @@ func (a Artifact) GetData() ([]byte, error) {
...
 
 // Save artifact to a specific path, using your own filename.
-func (a Artifact) Save(path string) (bool, error) {
-	data, err := a.GetData()
+func (a Artifact) Save(ctx context.Context, path string) (bool, error) {
+	data, err := a.GetData(ctx)
 
 	if err != nil {
-		return false, errors.New("No Data received, not saving file.")
+		return false, errors.New("no Data received, not saving file")
```
Comment (ZihanJiang96) [inline artifact.go:58]:
```suggestion
		return false, errors.New("no data received, not saving file")
```

File: build.go [hunk 2]
```diff
@@ -266,8 +267,8 @@ func (b *Build) GetConsoleOutputFromIndex(startID int64) (consoleResponse, error
 	return console, err
 }
 
-func (b *Build) GetCauses() ([]map[string]interface{}, error) {
-	_, err := b.Poll()
+func (b *Build) GetCauses(ctx context.Context) ([]map[string]interface{}, error) {
+	_, err := b.Poll(ctx, 3)
```
Comment (ZihanJiang96) [inline build.go:271]: why we need `3` here?

**Review by ZihanJiang96 — COMMENTED** [review]
just 2 minor things. 👍

File: build.go (hunk 2 above)
Comment (figo) [reply build.go]: not needed, removed

File: artifact.go (hunk 1 above)
Comment (figo) [reply artifact.go]: fixed, thanks

**Pull request merged**